DscConfigurationData/Private/Get-FileProviderData.ps1

```diff
@@ -0,0 +1,25 @@
+#Requires -module powershell-yaml
+#Using Module Datum
+
+function Get-FileProviderData {
+    [CmdletBinding()]
+    Param(
+        $Path,
+
+        [AllowNull()]
+        $DataOptions
+    )
+    foreach ($File in (get-childitem -Path $Path -File)) {
+        $FilePath = $File.FullName
+        Write-Verbose "Getting File Provider Data for Path: $FilePath"
+        switch ($File.Extension) {
+            '.psd1' { Import-PowerShellDataFile $FilePath | ConvertTo-Hashtable}
+            '.json' { Get-Content -Raw $FilePath | ConvertFrom-Json | ConvertTo-Hashtable }
+            '.yml' { convertfrom-yaml (Get-Content -raw $FilePath) | ConvertTo-Hashtable }
+            '.ejson' { Get-Content -Raw $FilePath | ConvertFrom-Json | ConvertTo-ProtectedDatum -UnprotectOptions $DataOptions}
```

Comment:
since we don't do everything that Datum does, I think we should remove the lines for ejson, eyaml and epsd1.